server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -66,7 +70,140 @@
         Property.NodeScope
     );
 
-    public static final ByteSizeValue DEFAULT_MAX_BYTES_PER_SEC = new ByteSizeValue(40L, ByteSizeUnit.MB);
+    /**
+     * Disk's write bandwidth allocated for this node. This bandwidth is expressed for write operations that have the default block size of
+     * {@link #DEFAULT_CHUNK_SIZE}.
+     */
+    public static final Setting<ByteSizeValue> NODE_BANDWIDTH_RECOVERY_DISK_WRITE_SETTING = bandwidthSetting(
+        "node.bandwidth.recovery.disk.write"
+    );
+
+    /**
+     * Disk's read bandwidth allocated for this node. This bandwidth is expressed for read operations that have the default block size of
+     * {@link #DEFAULT_CHUNK_SIZE}.
+     */
+    public static final Setting<ByteSizeValue> NODE_BANDWIDTH_RECOVERY_DISK_READ_SETTING = bandwidthSetting(
+        "node.bandwidth.recovery.disk.read"
+    );
+
+    /**
+     * Network's read bandwidth allocated for this node.
+     */
+    public static final Setting<ByteSizeValue> NODE_BANDWIDTH_RECOVERY_NETWORK_SETTING = bandwidthSetting(
+        "node.bandwidth.recovery.network"
+    );
+
+    static final double DEFAULT_FACTOR_VALUE = 0.4d;
+
+    /**
+     * Default factor as defined by the operator.
+     */
+    public static final Setting<Double> NODE_BANDWIDTH_RECOVERY_OPERATOR_FACTOR_SETTING = operatorFactorSetting(
+        "node.bandwidth.recovery.operator.factor",
+        DEFAULT_FACTOR_VALUE
+    );
+
+    public static final Setting<Double> NODE_BANDWIDTH_RECOVERY_OPERATOR_FACTOR_WRITE_SETTING = factorSetting(
+        "node.bandwidth.recovery.operator.factor.write",
+        NODE_BANDWIDTH_RECOVERY_OPERATOR_FACTOR_SETTING
+    );
+
+    public static final Setting<Double> NODE_BANDWIDTH_RECOVERY_OPERATOR_FACTOR_READ_SETTING = factorSetting(
+        "node.bandwidth.recovery.operator.factor.read",
+        NODE_BANDWIDTH_RECOVERY_OPERATOR_FACTOR_SETTING
+    );
+
+    public static final Setting<Double> NODE_BANDWIDTH_RECOVERY_OPERATOR_FACTOR_MAX_OVERCOMMIT_SETTING = Setting.doubleSetting(
+        "node.bandwidth.recovery.operator.factor.max_overcommit",
+        100d, // high default overcommit
+        1d,
+        Double.MAX_VALUE,
+        Property.NodeScope,
+        Property.OperatorDynamic
+    );
+
+    /**
+     * Default factor as defined by the user.
+     */
+    public static final Setting<Double> NODE_BANDWIDTH_RECOVERY_FACTOR_SETTING = factorSetting(
+        "node.bandwidth.recovery.factor",
+        NODE_BANDWIDTH_RECOVERY_OPERATOR_FACTOR_SETTING
+    );
```

Comment:
Does this setting do anything now?

Reply:
yeah, the overidding with user/op then read/write setting is a bit complex. Maybe I should just remove this one.

Reply:
Sounds good to me to remove it for now, we can re-add it in a follow-up doing the necessary plumbing for it.

Reply:
I pushed 95c6ccffe5bdf40dcab8a65e8b7c9f871f6c874c to remove the setting for now